mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java

```diff
@@ -67,6 +79,36 @@ protected void handleField(ResultSet resultSet, StructuredRecord.Builder recordB
       }
     }
     switch (sqlType) {
+      case Types.TIMESTAMP:
+        // SmallDateTime case
+        GregorianCalendar gc = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
+        gc.setGregorianChange(new Date(Long.MIN_VALUE));
+        Timestamp timestampSmalldatetime = resultSet.getTimestamp(columnIndex, gc);
+        if (timestampSmalldatetime != null) {
+          ZonedDateTime zonedDateTime = OffsetDateTime.of(timestampSmalldatetime.toLocalDateTime(),
+                                                          OffsetDateTime.now().getOffset())
+                                                          .atZoneSameInstant(ZoneId.of("UTC"));
+          if (Schema.LogicalType.DATETIME.equals(logicalType)) {
+            // SmallDateTime to CDAP DateTime type conversion
+            recordBuilder.setDateTime(field.getName(), zonedDateTime.toLocalDateTime());
```

Comment:
Why do we need this code ?

Reply:
removed this. Didn't notice that this case was already covered above